jira_connector.py

```diff
@@ -491,6 +492,33 @@ def _list_projects(self, param):
 
         return action_result.set_status(phantom.APP_SUCCESS)
 
+    def _get_base_url_from_url_path(self, url):
+        parsed_url = urlparse(url)
+        return f"{parsed_url.scheme}://{parsed_url.netloc}"
+
+    def _update_base_url_in_url_path(self, url, new_base):
+        parsed_url = urlparse(url)
+        new_parsed_base = urlparse(new_base)
+        return urlunparse((
+            new_parsed_base.scheme, new_parsed_base.netloc,
+            parsed_url.path, parsed_url.params,
+            parsed_url.query, parsed_url.fragment
+        ))
+
+    def _validate_and_update_custom_fields_url(self, custom_fields):
+        try:
+            allowed_values = custom_fields['custom_field']['allowed_values']
+            for value in allowed_values:
+                if 'self' not in value:
+                    continue
+                self_url = value['self']
+                base_url = self._get_base_url_from_url_path(self_url)
+                if base_url != self._base_url:
+                    value['self'] = self._update_base_url_in_url_path(self_url, self._base_url)
+            return custom_fields
+        except KeyError:
```

Comment:
do we need try except on the whole block?

we could move this try except out of ` def _validate_and_update_custom_fields_url(self, custom_fields):`

And then just do

```
try:
    custom_fields = self._validate_and_update_custom_fields_url(custom_fields)
except KeyError:
            return action_result.set_status(phantom.APP_ERROR, "Unable to validate custom fields"), None, None
```